[ONNX] Update legacy code, initialize onnx_shape_inference=True by default

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82040
* #82039
* #82038
* #82852
* __->__ #82767

Legacy code has onnx_shape_inference=False by default, which is misleading
as every other export api sets it to True unless otherwise overriden by caller.
There is only two tests that need updating according to this change.
* test_utility_funs.py::test_constant_fold_shape. The resulting number of nodes
  in graph is increased by 1, due to that previously the extra constant node was
  added as initializer.
* test_utility_funs.py::test_onnx_function_substitution_pass. Enabling onnx
  shape inference discovered discrepancy in test input shape and supplied dynamic
  axes arguments.

BowenBao committed:
[ONNX] Update legacy code, initialize onnx_shape_inference=True by default

Legacy code has onnx_shape_inference=False by default, which is misleading
as every other export api sets it to True unless otherwise overriden by caller.
There is only two tests that need updating according to this change.
* test_utility_funs.py::test_constant_fold_shape. The resulting number of nodes
  in graph is increased by 1, due to that previously the extra constant node was
  added as initializer.
* test_utility_funs.py::test_onnx_function_substitution_pass. Enabling onnx
  shape inference discovered discrepancy in test input shape and supplied dynamic
  axes arguments.

[ghstack-poisoned]

justinchuby commented on torch/onnx/symbolic_helper.py:
```diff
@@ -1364,7 +1364,7 @@ def _set_operator_export_type(operator_export_type):
 # This function is for debug use only.
-# onnx_shape_inference = False by default.
+# onnx_shape_inference = True by default.
 def _set_onnx_shape_inference(onnx_shape_inference: bool):
```
Is this function still used elsewhere? Can we remove it?

BowenBao replied:
I think only place left is `_export` api, where `onnx_shape_inference` is one of the argument. So can be altogether removed after its deprecation.

BowenBao committed:
Update on "[ONNX] Update legacy code, initialize onnx_shape_inference=True by default"


Legacy code has onnx_shape_inference=False by default, which is misleading
as every other export api sets it to True unless otherwise overriden by caller.
There is only two tests that need updating according to this change.
* test_utility_funs.py::test_constant_fold_shape. The resulting number of nodes
  in graph is increased by 1, due to that previously the extra constant node was
  added as initializer.
* test_utility_funs.py::test_onnx_function_substitution_pass. Enabling onnx
  shape inference discovered discrepancy in test input shape and supplied dynamic
  axes arguments.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update legacy code, initialize onnx_shape_inference=True by default"


Legacy code has onnx_shape_inference=False by default, which is misleading
as every other export api sets it to True unless otherwise overriden by caller.
There is only two tests that need updating according to this change.
* test_utility_funs.py::test_constant_fold_shape. The resulting number of nodes
  in graph is increased by 1, due to that previously the extra constant node was
  added as initializer.
* test_utility_funs.py::test_onnx_function_substitution_pass. Enabling onnx
  shape inference discovered discrepancy in test input shape and supplied dynamic
  axes arguments.

[ghstack-poisoned]

BowenBao commented:
cc @abock for merge approval

BowenBao commented:
@pytorchbot merge -g

BowenBao committed:
Update on "[ONNX] Update legacy code, initialize onnx_shape_inference=True by default"


Legacy code has onnx_shape_inference=False by default, which is misleading
as every other export api sets it to True unless otherwise overriden by caller.
There is only two tests that need updating according to this change.
* test_utility_funs.py::test_constant_fold_shape. The resulting number of nodes
  in graph is increased by 1, due to that previously the extra constant node was
  added as initializer.
* test_utility_funs.py::test_onnx_function_substitution_pass. Enabling onnx
  shape inference discovered discrepancy in test input shape and supplied dynamic
  axes arguments.

[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge -g
